src/clientInterface.ts

```diff
@@ -37,6 +37,9 @@ export interface ClientOptions
   clientConfig?: gax.ClientConfig;
   fallback?: boolean | 'rest' | 'proto';
   apiEndpoint?: string;
+  // clientCertSource must return a tuple of cert and key, which
+  // are used to configure mTLS:
```

Comment:
This method will actually be invoked in generated code.

Reply:
Is workflow like this: a user can pass `clientOptions` with `clientCertSource` defined to construct a client. Then generator invoke this callback and to get `cert` and `key`.

Is that all generator need to do to support mTLS?

Reply:
@summer-ji-eng I'll share the AIP. There's a couple environment variables we check for, and a well known file location (which might have information for connecting through mTLS).

I think this would make more sense in the generated code, potentially, than this layer?

Reply:
At the first glance, I think one possible design is:

For microgenerator, in the [client initialization](https://github.com/googleapis/nodejs-kms/blob/master/src/v1/key_management_service_client.ts#L250)
(1) check GOOGLE_API_USE_CLENT_CERTIFICATE env var. 
- If it is not true, set _opts.cert = _opts.key = null
- If it is true, there are 2 cases. If client_options.client_cert_source is not null, run it to produce cert/key; if it is null, then produce the default cert/key if exists. After that set _opts.cert and _opts.key. I think gax should provide APIs to check if default cert/key exists and produce them, since this is common code among all clients. 

(2) check GOOGLE_API_USE_MTLS_ENDPOINT env var. Figure out the api endpoint to use, based on if user provided api endpoint via _opts.apiEndpoint, and this env var value, and if _opts.cert, _opts.key exists. Then set _opts.service_path to the right endpoint (regular or mtls)

For gax, use the code in this PR + the APIs for default cert/key